Consul/Interfaces/IAgentEndpoint.cs

```diff
@@ -63,6 +63,7 @@ public interface IAgentEndpoint
         Task<QueryResult<LocalServiceHealth>> GetLocalServiceHealthByID(string serviceID, QueryOptions q, CancellationToken ct = default);
         Task<QueryResult<LocalServiceHealth>> GetLocalServiceHealthByID(string serviceID, CancellationToken ct = default);
         Task<QueryResult<Metrics>> GetAgentMetrics(CancellationToken ct = default);
+        Task<WriteResult<AgentAuthorizeResponse>> ConnectAuthorize(AgentAuthorizeParams parameters, CancellationToken ct = default);
```

Comment:
We can also add an overload with query options.

Reply:
Oh yeah, that makes a lot of sense. Haven't seen that pattern in code with WriteResult so it didn't occur to me. Will implement.

Reply:
Yep yep, that's what I thought. Just pushed the commit, waiting for the tests to run!